### PR TITLE
calibre: update to 7.19.0, add missing dependencies

### PR DESCRIPTION
**Pull request by stdmnpkg**
Topic Description
-----------------

- calibre: update to 7.19.0
- pyqt6-webengine: new, 6.7.0
- jxrlib: new, 0.2.4
- unrardll: new, 0.1.7
- lxml-html-clean: new, 0.2.2
- pychm: new, 0.8.6
- python-xxhash: new, 3.5.0

Package(s) Affected
-------------------

- calibre: 7.19.0

Security Update?
----------------

No

Build Order
-----------

```
#buildit calibre
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`

### CODE REVIEW

**Review by eatradish — DISMISSED** [review]
Missing `.desktop` file

**Comment by stdmnpkg** [comment]
now require MathJax 3, meanwhile bundled MathJax can be used

- [ ] mathjax

according to Arch Package Database and Fedora Packages, v7.16.0 additionally require following packages:

- [x] jxrlib
- [x] python-lxml-html-clean
- [x] python-pychm
- [x] python-xxhash
- [x] python-pyqt6-webengine
- [x] python-unrardll

these are probably also required

- [ ] python-faust-cchardet
- [ ] python-pdftotext
- [ ] python-py7zr
- [ ] rapydscript-ng (build only)

**Comment by cthbleachbit** [comment]
Any chances we can have calibre install a `.desktop` file in `/usr/share/applications`? Neither stable nor this topic installs one - at this point calibre has to be started from command line.

**Comment by stdmnpkg** [comment]
@cthbleachbit There will be a desktop file. I will fix it after usability problem fixed.

File: runtime-imaging/jxrlib/autobuild/prepare [hunk 1]
```diff
@@ -0,0 +1 @@
+cp "$SRCDIR"/autobuild/CMakeLists.txt "$SRCDIR"/CMakeLists.txt
```
Comment (MingcongBai) [inline runtime-imaging/jxrlib/autobuild/prepare]: abinfo and verbosity here.

File: lang-python/lxml-html-clean/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,6 @@
+PKGNAME=lxml-html-clean
+PKGSEC=python
+PKGDEP="python-3 lxml"
+PKGDES="HTML cleaner from lxml project"
+
+ABSPLITDBG=0
```
Comment (MingcongBai) [inline lang-python/lxml-html-clean/autobuild/defines]: Why?

File: runtime-imaging/jxrlib/autobuild/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# from https://src.fedoraproject.org/rpms/jxrlib/blob/rawhide/f/jxrlib.spec
```
Comment (MingcongBai) [inline runtime-imaging/jxrlib/autobuild/CMakeLists.txt]: Why did we replace a whole CMakeLists.txt?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: runtime-imaging/jxrlib/autobuild/CMakeLists.txt (hunk 3 above)
Comment (stdmnpkg) [reply runtime-imaging/jxrlib/autobuild/CMakeLists.txt]: original project's Makefile attempt build a Java binding which is not required for now, has no way to disable and fail to build here.

a tracking repo has been created at https://github.com/AOSC-Tracking/jxrlib with this change.

Comment (stdmnpkg) [reply runtime-imaging/jxrlib/autobuild/CMakeLists.txt]: CMakeLists.txt itself is originally from https://jxrlib.codeplex.com/workitem/13 (archive at https://codeplexarchive.org/codeplex/projecttab/issues/jxrlib/13), which says:

> # Patch for warnigns + CMake for building
> The following changes were made for the Fedora package which I'm proposing upstream:
> 
> - jxrlib_warnings.patch: Fix various warnings 
> - CMakeLists.txt: use CMake for building, to facilitate the creation of shared libraries, as well as cross-compilation. Debian does similarly.
>

File: lang-python/python-xxhash/autobuild/defines [hunk 4]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=python-xxhash
+PKGSEC=python
+PKGDEP=""
```
Comment (MingcongBai) [inline lang-python/python-xxhash/autobuild/defines]: Add python-3 dep here.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Comment by MingcongBai** [comment]
Does not launch as of the latest revision:

图片

**Review by MingcongBai — APPROVED** [review]
LGTM on AMD64:

图片

**Pull request merged**